src/main/java/repository/InteractionWithDB.java

```diff
@@ -0,0 +1,116 @@
+package repository;
+
+import model.CarEntity;
+
+import java.sql.*;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class InteractionWithDB {
+    Logger logger = Logger.getLogger(getClass().getName());
+    public Connection connect(String dataName, String user, String pass){
```

Comment:
metoda jest dobra tylko zrób to inaczej, samo utworzenie tego Repo powinno inicjować połączenie do bazy, daj dataName, user, password jako prywatne finalne, inicjalizację ich daj w konstruktrze. Za każdym razem jak stworzysz instancję tej klasy to będziesz miał już połączenie do bazy. Warto by zadbać żeby klasa była również Singletonem, żebyś miał tylko jeden taki obiekt w swojej apce https://refactoring.guru/design-patterns/singleton/java/example